polygeist/tools/cgeist/Lib/ValueCategory.cc

```diff
@@ -109,9 +109,7 @@ void ValueCategory::store(mlir::OpBuilder &builder, mlir::Value toStore) const {
     // Ad-hoc extension of booleans
     auto ElementType =
         llvm::TypeSwitch<Type, Type>(val.getType())
-            .Case<LLVM::LLVMPointerType>(
-                [](auto Ty) -> Type { return Ty.getElementType(); })
-            .Case<MemRefType>(
+            .Case<MemRefType, LLVM::LLVMPointerType>(
                 [](auto Ty) -> Type { return Ty.getElementType(); })
             .Default([](auto) -> Type { llvm_unreachable("Unhandled type"); });
```

Comment:
Remove the default just like in other places ?

Reply:
Missed that one. Fixed.